### ISSUE TITLE
Don't explicitly initialize variable to null false positive

### ISSUE THREAD

**Issue by xavierhainaux** [dart-lang/linter#210]
`final` and `const` variable must be initialized (or the analyzer produce an error).

This code should not produces the "Don't explicitly initalized variable to null" lint because we cannot fix it.
```dart
final String a = null;
const b = null;

class A {
  final int a = null;
  static const b = null;
}
```

**Comment by zoechi** [comment]
Isn't this quite uncommon? I guess I'd prefer to add an `// ignore: ...` comment to make it explicit that this is intentional.

**Comment by xavierhainaux** [comment]
It is probably uncommon but it could happen.

When the variable exist to conform to an interface, we could probably replace it with a getter like
```dart
  String get a => null;
```

But I think we should not have a lint that cannot be fixed easily. 
And I don't know how to use the "ignore" comment (what is the "..."?)

**Comment by zoechi** [comment]
Sorry for not being more clear

```
      // ignore: AVOID_INIT_TO_NULL
      final String a = null;
```
See also the error codes that can be ignored https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/lib/src/generated/error.dart

That's a new feature added very recently.

**Comment by pq** [comment]
While uncommon, I agree that this is annoying (and easily remedied in the lint).  Thanks for the report!
